homeassistant/components/ipma/diagnostics.py

```diff
@@ -0,0 +1,32 @@
+"""Diagnostics support for IPMA."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+
+from .const import DATA_API, DATA_LOCATION, DOMAIN
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    location = hass.data[DOMAIN][entry.entry_id][DATA_LOCATION]
+    api = hass.data[DOMAIN][entry.entry_id][DATA_API]
+
+    return {
+        "location_information": {
+            "latitude": round(float(entry.data[CONF_LATITUDE]), 3),
+            "longitude": round(float(entry.data[CONF_LONGITUDE]), 3),
```

Comment:
I think we usually redact these

Reply:
Since everything works with stations, can't you use the station name to debug what you need?

Reply:
Can't... this is actual an important part of the debug, determining if the underlying library is picking up the right nearby station. To preserve some library I'm routing to 3 decimals 

Also, some stations share the same name for some hellish reason :(

Reply:
I do not and issue actually